### PR TITLE
Update docs about blocks styling to clearly state global styles are the recommended approach

### LINKED ISSUE #47238: Update WooCommerce Blocks documentation about theming to clearly state that global styles are the preferred way for styling blocks and block templates
We have some docs about theming WooCommerce blocks here:

https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce-blocks/docs/designers/theming

However, that doc hasn't been updated for a long time and it doesn't mention global styles, which is the new and recommended way to style blocks. It also puts a lot of emphasis on CSS class names, which we want to avoid being used by themes.

### PR DESCRIPTION
**Pull request by Aljullu**
### Changes proposed in this Pull Request:

Our docs about styling WooCommerce blocks were a bit outdated and made a lot of emphasis on CSS class names, without mentioning global styles, which is the preferred and recommended way to style blocks. This PR updates this. You can read pf59Ax-14H-p2 for more details.

Closes #47238.

### How to test the changes in this Pull Request:

_(No need to test when testing the release)_

1. Read the docs and verify they are correct and they are aligned with our recommendations to style blocks.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce-blocks/docs/designers/theming/README.md [hunk 1]
```diff
@@ -4,12 +4,53 @@ This page includes all documentation regarding WooCommerce Blocks and themes.
...
+
+* Better performance, as only the required CSS is printed into the page, reducing the bundle size to render a page.
+* Can be easily customized by users via the UI.
+* Gracefully handle conflicts between plugins and themes.
+* Are not affected by markup or class name updates into individual blocks or components.
+* Don't depend on a specific nesting order of blocks: users can freely move blocks around without styles breaking.
+
+#### Example
+
+For example, let's imagine you are building a theme and would like to customize the Product Price block styles, you can do so adding these properties in your `theme.json`:
```
Comment (nefeline) [inline plugins/woocommerce-blocks/docs/designers/theming/README.md]:
```suggestion
For example, let's imagine you are building a theme and would like to customize the Product Price block styles, you can do so by adding these properties in your `theme.json`:
```

File: plugins/woocommerce-blocks/docs/designers/theming/README.md [hunk 2]
```diff
@@ -4,12 +4,53 @@ This page includes all documentation regarding WooCommerce Blocks and themes.
...
+
+You can find more [documentation on global styles](https://developer.wordpress.org/themes/global-settings-and-styles/styles/) in developer.wordpress.org. You can also find the [list of WooCommerce blocks and their names in the docs](../../block-references/block-references.md).
+
 ### Block and component class names
 
+**Important: we strongly discourage writing CSS code based on existing block class names, prioritize using global styles when possible. We specially discourage writing CSS selectors that rely on a specific block being a descendant of another one, as users can move blocks around freely, so they are prone to breaking. Similar to WordPress itself, we consider the HTML structure within components, blocks, and block templates to be “private”, and subject to further change in the future, so using CSS to target the internals of a block or a block template is _not recommended or supported_.**
```
Comment (nefeline) [inline plugins/woocommerce-blocks/docs/designers/theming/README.md]:
```suggestion
**Important: we strongly discourage writing CSS code based on existing block class names and prioritize using global styles when possible. We especially discourage writing CSS selectors that rely on a specific block being a descendant of another one, as users can move blocks around freely, so they are prone to breaking. Similar to WordPress itself, we consider the HTML structure within components, blocks, and block templates to be “private”, and subject to further change in the future, so using CSS to target the internals of a block or a block template is _not recommended or supported_.**
```

**Review by nefeline — COMMENTED** [review]
Nice work! Just noticed a few typos along the way :)

**Comment by Aljullu** [comment]
Good catches! I just merged both suggestions, thanks for the review, @nefeline!

**Review by nefeline — APPROVED** [review]
Awesome, thanks!

**Pull request merged**